src/world/common/UnkNpcAIFunc31.inc.c

```diff
@@ -0,0 +1,18 @@
+#include "common.h"
+#include "map.h"
+#include "effects.h"
+
+void N(UnkNpcAIFunc31)(Evt* script, NpcAISettings* aiSettings, EnemyTerritoryThing* territory) {
+    Enemy* enemy = script->owner1.enemy;
+    Npc* npc = get_npc_unsafe(enemy->npcID);
+
+    if (npc->duration > 0) {
+        npc->duration--;
+        return;
```

Comment:
I think this would be better as an else instead of early return, and you should be able to do animList[0] below